tests/unit/test_data.py

```diff
@@ -9,3 +15,46 @@ def test_repeating_loader():
         assert next(loader) == 1
         assert next(loader) == 2
         assert next(loader) == 3
+
+
+@pytest.mark.parametrize('train_batch_size, drop_last',
+                         [(1,
+                           True),
+                          (4,
+                           True),
+                          (1,
+                           False),
+                          (4,
+                           False)])
+def test_dataloader_drop_last(tmpdir, train_batch_size, drop_last):
+    config_dict = {
+        "train_batch_size": train_batch_size,
+        "dataloader_drop_last": drop_last,
+        "steps_per_print": 1,
+        "fp16": {
+            "enabled": True
+        },
+    }
+    args = args_from_dict(tmpdir, config_dict)
+    hidden_dim = 10
+
+    model = SimpleModel(hidden_dim)
+
+    @distributed_test(world_size=[1])
+    def _test_dataloader_drop_last(args, model, hidden_dim):
+        optimizer = torch.optim.AdamW(params=model.parameters())
+        batch_size = model.train_micro_batch_size_per_gpu()
```

Comment:
This line is failing unit tests because SimpleModel does not have this attribute.
![image](https://user-images.githubusercontent.com/4271600/131523156-4292efec-855f-4f36-a236-cdaf8f82b486.png)

A solution is to pass batch into `_test_dataloader_drop_last()`.

Reply:
On a second look, it seems you don't need `batch_size` at all, am I correct? I think you can just delete that line.

Reply:
@tjruwase 

I have a few doubts.

1. Should I declare the `batch_size = N` or should I use the `model.train_micro_batch_size_per_gpu()`, which seems to be not working since `SimpleModel.py` doesn't have that function.
2.  If I statically declare the `batch_size` and pass It to `_test_dataloader_drop_last()`, where do I have to use it inside the function? since all the functions inside it don't have such a parameter.

Thanks!

Reply:
> On a second look, it seems you don't need `batch_size` at all, am I correct? I think you can just delete that line.

Sure! I was wondering the same. My bad.

Thank you!

Reply:
I don't think you need to use `batch_size` at all since you are not constructing a dataloader, DeepSpeed will do that for you. DeepSpeed knows how to extract `batch_size` from the json config as you see [here](https://github.com/microsoft/DeepSpeed/blob/ddffbae021fef24bf64945d45ace9bac7e08508d/deepspeed/runtime/engine.py#L1179).

Reply:
Got it! I am learning a lot. Thank you for your patience and support,